example/passthrough.c

```diff
@@ -144,9 +148,34 @@ static int xmp_mknod(const char *path, mode_t mode, dev_t rdev)
 		res = open(path, O_CREAT | O_EXCL | O_WRONLY, mode);
 		if (res >= 0)
 			res = close(res);
-	} else if (S_ISFIFO(mode))
+	} else if (S_ISFIFO(mode)) {
 		res = mkfifo(path, mode);
-	else
+#ifdef __FreeBSD__
```

Comment:
Let's put all this code into a separate function, so that the callback remains easy to understand. I'm thinking something like `mknod_wrapper`, which for Linux just calls mknod and for FreeBSD does the extra stuff.